### ISSUE TITLE
Features

### ISSUE THREAD

**Issue by CTSchorsch** [SolidGeek/VescUart#2]
Hi,

your lib works really fine. Thank you for your work. 

Do you think it is possible to read the Temperature and the drv error too?

Greetings 
Georg

**Comment by SolidGeek** [comment]
Yeah that should be quite easy. I am actually skipping the first 4 bytes of the data received, which contains the mosfet temperature (2 bytes) and the motor temperature (2 bytes). Regarding the error you are looking for, I can get you the mc_fault_code, which can hold any of these values:

`typedef enum {
	FAULT_CODE_NONE = 0,
	FAULT_CODE_OVER_VOLTAGE,
	FAULT_CODE_UNDER_VOLTAGE,
	FAULT_CODE_DRV,
	FAULT_CODE_ABS_OVER_CURRENT,
	FAULT_CODE_OVER_TEMP_FET,
	FAULT_CODE_OVER_TEMP_MOTOR
} mc_fault_code;`

I can implement it in a dev branch if you want to try it out?

**Comment by CTSchorsch** [comment]
Hi,

yes. This would be fine. I use your lib to write the housekeeping data to a sdcard and this values are missing :)

Georg

**Comment by CTSchorsch** [comment]
some error in your lib


error: invalid conversion from 'uint8_t {aka unsigned char}' to
'mc_fault_code' [-fpermissive]
    data.error     = message[ind];
                                ^
exit status 1



Am 28.11.18 um 22:00 schrieb SolidGeek:

**Comment by SolidGeek** [comment]
I have changed the datatype of the error variable to uint8_t, should work now. Its weird, because I didn't get the error on compiling. Would you give it a try again? Just redownload the development branch 👍

**Comment by CTSchorsch** [comment]
Yes, thanks, now its working (compiling and values are fine)
The error value is one from the mc_fault_code ENUM ?

**Comment by SolidGeek** [comment]
Glad to hear. Yeah the error value map directly to the mc_fault_code ENUM. So an error value of 3 would mean a DVR error etc.

**Comment by CTSchorsch** [comment]
Perfect :)

**Comment by vicgwee** [comment]
Is there a possibility to reduce the time taken by getVescValues()? 

One option is to increase the baud rate. On my Mega, I've recorded the following times taken (averaged over 10,000 tries): 

Baud Rate | Time taken (ms) 
-- | -- 
9600 | 79.04  
40000 | 19.30 
115200 | 6.98 
200000 | 4.23 
250000 | 3.46   

*Over 400,000 I begin to encounter some communication failures.

Wondering what's the bottleneck: unpacking the payload? or receiving the payload from VESC.

What if we could separate the function into two: 
getVescValues() -- send the request to the VESC
checkVescValues() -- check the Serial buffer if the payload has been received, if so then unpack the payload and update the values of .data

Thus the code doesn't block the arduino while waiting to receive the payload.

**Comment by daniel84uk** [comment]
Thanks for this, i spent ages trying to figure out how to read the bloody temps lol.
